.github/workflows/proto-generate-documentation.yml

```diff
@@ -0,0 +1,33 @@
+name: "Proto: Generate Documentation"
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - proto/**
+
+jobs:
+  generateProtoDocumentation:
+    name: "Generate Protobuf Documentation"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.18'
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+      - name: Generate Docs
+        run: |
+          go install github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc@latest
+          which protoc-gen-doc
+          which protoc
+          cd proto/
+          mkdir -p ./docs/schema
+          protoc --doc_out=./docs/schema --doc_opt=markdown,docs.md ./serverless/instrumentation/tags/v1/*.proto ./serverless/instrumentation/v1/*.proto
+      - name: Add Doc Commit
+        uses: EndBug/add-and-commit@v9
```

Comment:
How will it behave if there are no changes in the generated doc?

Reply:
It will not commit if the diff is the same. That being said, we can change it that the workflow only runs against the folder that contains `.proto` files. That means it will always be a change.

Reply:
>  That means it will always be a change.

There may be just whitespace changes or changes to comments that I believe will not influence outcome doc.

Still if no empty commit will be pushed and CI will not crash it's ok